src/core/control/settings/Settings.h

```diff
@@ -569,7 +570,8 @@ class Settings {
     void setStabilizerAveragingMethod(StrokeStabilizer::AveragingMethod averagingMethod);
     void setStabilizerPreprocessor(StrokeStabilizer::Preprocessor preprocessor);
 
-    const Palette& getColorPalette();
+    const std::optional<fs::path> getColorPaletteSetting();
```

Comment:
```suggestion
    auto getColorPaletteSetting() -> fs::path const&;
```
Returning a const value may force a double copy, the copy won't be const anymore.
Returning an optional to an implicitly on the heap allocated value is over verbose, since an empty path would be the same.